apps/aecore/lib/aecore/chain/block_validation.ex

```diff
@@ -176,4 +179,28 @@ defmodule Aecore.Chain.BlockValidation do
   defp check_correct_height?(new_block, previous_block) do
     previous_block.header.height + 1 == new_block.header.height
   end
+
+  @spec valid_header_timestamp(Block.t()) :: boolean()
+  defp valid_header_timestamp(%Block{header: new_block_header}) do
+    case new_block_header.timestamp <= :os.system_time() + 3_600_000 do
+      true ->
+        top_block = Chain.top_block()
```

Comment:
does this work? we had issues with calling the chain for fetching blocks to calculate the next difficulty and passed `blocks_for_difficulty_calculation`